components/datatypes/consentpreferences.schema.json

```diff
@@ -0,0 +1,424 @@
+            "xdm:phoneCalls": {
+              "title": "Receive phone calls",
+              "description": "The company may contact the user with phone solicitations",
+              "$ref": "#/definitions/marketingOption"
+            },
+            "xdm:physicalMail": {
+              "title": "Receive physical mail",
+              "description": "It is okay to send offers via traditional/physical mail.",
+              "$ref": "#/definitions/marketingOption"
+            },
+            "xdm:inVehicleMessages": {
+              "title": "Receive in-vehicle messages",
+              "description": "It is okay to provide offers to the user via technology installed in their vehicle",
+              "$ref": "#/definitions/marketingOption"
+            },
+            "xdm:inHomeMessages": {
+              "title": "Receive in-home messages",
+              "description": "It is okay to provide offers to the user via technology installed in their home",
+              "$ref": "#/definitions/marketingOption"
+            },
+            "xdm:iotMessages": {
+              "title": "Receive IoT messages",
+              "description": "It is okay to present offers via less traditional types of connected devices",
+              "$ref": "#/definitions/marketingOption"
+            },
+            "xdm:socialMedia": {
+              "title": "Receive Social Media Messages",
+              "description": "It is okay to direct messages to the user via their social media accounts",
+              "$ref": "#/definitions/marketingOption"
+            }
+          }
+        }
+      }
+    },
+    "consent-metadata": {
+      "properties": {
+        "xdm:version": {
+          "title": "Preferences Version",
+          "description": "Version of the Privacy Preferences Standard",
+          "type": "string",
+          "pattern": "^[0-9]{1,2}\\.[0-9]{1,2}\\.[0-9]{1,4}$",
+          "default": "1.0.0"
+        },
+        "xdm:timestamp": {
+          "title": "Consent Preferences timestamp",
+          "description": "Timestamp of the complete set of user consent preferences",
+          "type": "string",
+          "format": "date-time"
+        },
+        "xdm:source": {
+          "title": "Consent Preferences source",
+          "description": "Source for the complete set of user consent preferences",
+          "$ref": "#/definitions/source"
+        },
+        "xdm:sourceID": {
+          "title": "?? Source ID",
+          "description": "ID for user as defined by consent source",
+          "type": "string",
+          "maxLength": 20
+        },
+        "xdm:userCountry": {
+          "title": "User Country/Region",
+          "description": "ISO 3166 alpha2 country code or ISO 3166-2 country plus state/province code for use in helping to determine how to interpret consent values",
+          "type": "string",
+          "maxLength": 6,
+          "pattern": "[A-Z]{2}(-[A-Z]{1,3}){0,1}"
+        },
+        "xdm:countrySource": {
```

Comment:
Will this account for regions within the same country (say CA) or say regions across countries (say EU) in case they have different consent regulations? If yes, then should we rename this to reflect that?

Reply:
ISO 3166 alpha2 country code is a two character code identifying the country.
ISO 3166-2 adds a 1 to 3 character code to identify regions, states, provinces (and in some cases cities) within the country. For the US it is a 2 character state code. 
So customer can use "US" if they will apply the same regulations for all US customers, or "US-CA", "US-NY", ... if they plan to treat California customers differently than others in the US, because of California's stricter regulations. Or they can use "US-CA", ... and only look at the first two characters to treat all users in the US the same.
Do you have a suggestion for an alternate name? userCountryRegion? userCountryOptionalRegion? Country is accurate, because it always identifies the country, but sometimes it can be more precise. userCountryPlus?

Reply:
Do we need EU, EU-Italy, EU-Italy-Tuscany?

Reply:
ISO 3166 does not include "EU" as a country code. To see if GDPR applies you need to check if this field contains the country code for any country in the EU. Note that the code for Tuscany is "IT-52". I will update the pattern to support digits for the region portion of the code.

Reply:
I am also renaming this field to userCountryRegionCode

Reply:
Sounds good